pkg/api/v1/types.go

```diff
@@ -710,6 +725,22 @@ type Container struct {
 	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
 }
 
+func (ctr *Container) ApplyDefaults() {
```

Comment:
And I think breaking up types.go into smaller pieces, fracturing the API surface into multiple groups, putting the registry (and ideally controller) implementations near the API types, and separating the API machinery from the API schema are more important. I want adding a new API type to be pretty self-contained and modular.